src/fileformats/ocd_georef.cpp

```diff
@@ -0,0 +1,409 @@
+    { OCD_GRID_BRAZIL, 21, 31982 }, // SIRGAS 2000 / UTM zone 22S
+    { OCD_GRID_BRAZIL, 22, 31983 }, // SIRGAS 2000 / UTM zone 23S
+    { OCD_GRID_BRAZIL, 23, 31984 }, // SIRGAS 2000 / UTM zone 24S
+    { OCD_GRID_BRAZIL, 24, 31985 }, // SIRGAS 2000 / UTM zone 25S
+    { OCD_GRID_BRITAIN, 0, 27700 }, // OSGB 1936
+    { OCD_GRID_CZECHIA, 1, 5514 }, // S-JTSK Krovak East North
+    { OCD_GRID_CZECHIA, 2, 28403 }, // Pulkovo 1942 / Gauss-Kruger zone 3
+    { OCD_GRID_FINLAND, 2, 2392 }, // KKJ / Finland zone 2
+    { OCD_GRID_FINLAND, 5, 3067 }, // TM35FIN, ETRS89 / ETRS-TM35FIN
+    { OCD_GRID_FINLAND, 12, 3879 }, // ETRS89 / GK25FIN
+    { OCD_GRID_FINLAND, 25, 3132 }, // ETRS89 / ETRS-GK25FIN
+    { OCD_GRID_GOOGLE_TMS, 0, 900913 }, // Google Mercator
+    { OCD_GRID_HUNGARY, 0, 23700 }, // HD72 / EOV
+    { OCD_GRID_ICELAND, 4, 3057 }, // ISN93 / Lambert 1993
+    { OCD_GRID_ISRAEL, 0, 2039 }, // Israel 1993 / Israeli TM Grid
+    { OCD_GRID_GERMANY, 2, 31466 }, // DHDN, Gauss-Krüger Zone 2
+    { OCD_GRID_GERMANY, 3, 31467 }, // DHDN, Gauss-Krüger Zone 3
+    { OCD_GRID_GERMANY, 4, 31468 }, // DHDN, Gauss-Krüger Zone 4
+    { OCD_GRID_GERMANY, 5, 31469 }, // DHDN, Gauss-Krüger Zone 5
+    { OCD_GRID_SLOVAKIA, 1, 5514 }, // S-JTSK Krovak East North
+    { OCD_GRID_SLOVENIA, 1, 3912 }, // MGI 1901 / Slovene National Grid
+    { OCD_GRID_SWEDEN, 2, 3006 }, // SWEREF99 TM
+    { OCD_GRID_SWEDEN, 3, 3007 }, // SWEREF99 12 00
+    { OCD_GRID_SWEDEN, 4, 3008 }, // SWEREF99 13 30
+    { OCD_GRID_SWEDEN, 5, 3009 }, // SWEREF99 15 00
+    { OCD_GRID_SWEDEN, 6, 3010 }, // SWEREF99 16 30
+    { OCD_GRID_SWEDEN, 7, 3011 }, // SWEREF99 18 00
+    { OCD_GRID_SWEDEN, 8, 3012 }, // SWEREF99 14 15
+    { OCD_GRID_SWEDEN, 9, 3013 }, // SWEREF99 15 45
+    { OCD_GRID_SWEDEN, 10, 3014 }, // SWEREF99 17 15
+    { OCD_GRID_SWEDEN, 11, 3015 }, // SWEREF99 18 45
+    { OCD_GRID_SWEDEN, 12, 3016 }, // SWEREF99 20 15
+    { OCD_GRID_SWEDEN, 13, 3017 }, // SWEREF99 21 45
+    { OCD_GRID_SWEDEN, 14, 3018 }, // SWEREF99 23 15
+    { OCD_GRID_SWITZERLAND, 1, 21781 }, // CH1903 / LV03 -- Swiss CH1903 / LV03
+    { OCD_GRID_SWITZERLAND, 2, 2056 }, // CH1903+ / LV95 -- Swiss CH1903+ / LV95
+};
+
+auto OcdGeoref::translateOcdMapper(const int ocd_grid_id, const int ocd_zone_id)
+{
+	std::vector<MapperCrsRecord> retval;
+
+	switch (ocd_grid_id)
+	{
+	case OCD_GRID_LOCAL:
+		retval.push_back({ MapperCrs::LOCAL, 0 });
+		return retval;
+	case OCD_GRID_UTM:
+		if (std::abs(ocd_zone_id) >= 1 && std::abs(ocd_zone_id) <= 60)
+		{
+			retval.push_back({ MapperCrs::UTM, ocd_zone_id });
+
+			if (ocd_zone_id > 0)
+				retval.push_back({ MapperCrs::EPSG, 32600 + ocd_zone_id });
+			else
+				retval.push_back({ MapperCrs::EPSG, 32700 - ocd_zone_id });
+		}
+		break;
+
+	case OCD_GRID_GERMANY:
+		// DHDN, Gauss-Krüger, Zone 2-5
+		if (ocd_zone_id >= 2 && ocd_zone_id <= 5)
+		{
+			retval.push_back({ MapperCrs::GAUSS_KRUEGER, ocd_zone_id });
+		}
+		[[ fallthrough ]]; // get the EPSG code
```

Comment:
No `[[ fallthrough ]]` with older GCC (Android). I didn't even dare to use Q_FALLTHROUGH, but this is something which we could add to our backports.

Reply:
Good point. Converted `[[ fallthrough ]];` to `// fallthrough` so that it can be converted back later on.